src/sentry_value.c

```diff
@@ -1124,10 +1124,32 @@ sentry_value_new_stacktrace(void **ips, size_t len)
     return stacktrace;
 }
 
+sentry_value_t
+sentry__value_new_span(sentry_value_t parent, const char *operation)
+{
+    sentry_value_t span = sentry_value_new_object();
+
+    sentry_transaction_context_set_operation(span, operation);
```

Comment:
I wonder if this should now be renamed `sentry_span_set_operation(span, operation)`.  Setting the operation on the transaction context is already a very nice API because you might as well pass the right value into the `sentry_value_new_transaction_context()` call.

Reply:
i think what we can do is define one common implementation for setting operations on spans/transactions/transaction contexts, but i'm thinking that the final API will most likely include a `sentry_span_set_operation(...)`, `sentry_transaction_set_operation(...)`, and `sentry_transaction_context_set_operation(...)`.

i think having a setter regardless can be useful in case one is only able to figure out what a transaction might be after a transaction context is created, or a transaction is started. it's not an immutable field until the transaction is finished.

regarding the naming and reusing `..._span_[operation]` for transactions and transaction contexts, i can't say that i'm too big of a fan of relying on the implicit knowledge that transactions are spans with extra bells and whistles in this SDK, since:
1. there's no such thing as "inheritance" in this to begin with
2. nothing in the types or the codebase hint at the idea that a transaction "is" a span

this might make sense for other SDKs where class/struct inheritance does exist, and transactions do indeed get `set_operation` and other methods for free by virtue of those extending the span implementation. the native SDK doesn't really programmatically support this. i also don't know if the user gains much from knowing that a transaction is loosely equivalent to a span.